homeassistant/components/imap/coordinator.py

```diff
@@ -92,7 +87,8 @@ async def _cleanup(self) -> None:
                 await self.imap_client.stop_wait_server_push()
                 await self.imap_client.logout()
                 await self.imap_client.close()
-            except AioImapException:
+            except (AioImapException, asyncio.TimeoutError) as ex:
+                self.async_set_update_error(ex)
                 _LOGGER.warning("Error while cleaning up imap connection")
```

Comment:
`self.imap_client.logout()` or `self.imap_client.close` could cause an `asyncio.TimeoutError`. We do not want cleanup to break the (background) task.